edgelet/iotedge/src/check/checks/latest_runtime_modules.rs

```diff
@@ -0,0 +1,182 @@
+use std::env::consts::{ARCH, OS};
+
+use failure::{self, Context, ResultExt};
+use os_info::Bitness;
+use serde_json::Value;
+
+use crate::check::{checker::Checker, Check, CheckResult};
+use crate::error::{DetermineEdgeVersionReason, ErrorKind};
+use crate::{DockerImageInfo, LatestVersions};
+
+#[derive(Default, serde_derive::Serialize)]
+pub(crate) struct LatestRuntimeModules {
+    pub actual_edge_agent_version: DockerImageInfo,
+    pub expected_edge_agent_version: DockerImageInfo,
+    pub actual_edge_hub_version: DockerImageInfo,
+    pub expected_edge_hub_version: DockerImageInfo,
+    pub latest_versions: crate::LatestVersions,
+}
+
+impl Checker for LatestRuntimeModules {
+    fn id(&self) -> &'static str {
+        "aziot-edge-version"
+    }
+
+    fn description(&self) -> &'static str {
+        "aziot-edge package is up-to-date"
+    }
+
+    fn execute(
+        &mut self,
+        check: &mut Check,
+        tokio_runtime: &mut tokio::runtime::Runtime,
+    ) -> CheckResult {
+        let docker_host_arg = if let Some(docker_host_arg) = &check.docker_host_arg {
+            docker_host_arg
+        } else {
+            return CheckResult::Skipped;
+        };
+
+        // Determine actual running versions of iotedge daemon, agent, and hub
+        self.actual_edge_agent_version.image_id =
+            match LatestRuntimeModules::actual_module_image_id(docker_host_arg, "edgeAgent") {
+                Ok(v) => v,
+                Err(e) => return CheckResult::Failed(e),
+            };
+        // TODO: Consider ignoring errors caused by EdgeHub not currently running
+        self.actual_edge_hub_version.image_id =
+            match LatestRuntimeModules::actual_module_image_id(docker_host_arg, "edgeHub") {
+                Ok(v) => v,
+                Err(e) => return CheckResult::Failed(e),
+            };
+
+        if check.expected_aziot_edge_agent_image_id.is_none()
+            || check.expected_aziot_edge_hub_image_id.is_none()
+        {
+            if check.parent_hostname.is_some() {
+                // This is a nested Edge device so it may not be able to access aka.ms or github.com.
+                // In the best case the request would be blocked immediately, but in the worst case it may take a long time to time out.
+                // The user didn't provide the `expected_aziot_edged_version` param on the CLI, so we just ignore this check.
+                return CheckResult::Ignored;
+            }
+
+            self.latest_versions = match LatestVersions::get_latest_versions(
+                tokio_runtime,
+                "https://aka.ms/AAdb3gt",
```

Comment:
This aka.ms URL was created for testing against the new version file format. Before merging, it should be changed to https://aka.ms/latest-aziot-edge and the following file that it points to, should be updated to use the new format:

https://raw.githubusercontent.com/Azure/azure-iotedge/master/latest-aziot-edge.json